src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java

```diff
@@ -0,0 +1,89 @@
+package juja.sqlcmd;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InMemoryDatabaseManager implements DatabaseManager {
+    private static final String NAME_DB_TEST = "sqlcmd_test";
+    private static final String LOGIN_FOR_TEST = "sqlcmd";
+    private static final String PASSWORD_FOR_TEST = "sqlcmd";
+
+    private boolean connected = false;
+    private HashMap<String, DataSet[]> tables = new HashMap<>();
+
+
+    @Override
+    public boolean connect(String database, String user, String password) {
+        connected = NAME_DB_TEST.equals(database) && LOGIN_FOR_TEST.equals(user) && PASSWORD_FOR_TEST.equals(password);
+        return connected;
+    }
+
+    @Override
+    public boolean insert(String tableName, DataSet dataset) {
+        DataSet[] newDataSets = tables.get(tableName);
+        if ((newDataSets != null)) {
+            if ((newDataSets.length == 0)
+                    || ((newDataSets[0].row().split(",").length >= dataset.row().split(",").length))) {
```

Comment:
I think instead `dataset.row().split(",").length)` you can add new method to Dataset.
This method should return number of  column.
After that your code will be clear.

and I agree with Shtramak you should think about your if condition, it difficult read now

Reply:
Thanks, i will do it